bin/zlock

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python
+#
+# zlock - lock a ZFS filesystem from being replicated by zbackup.
+#
+# In fact, the parameter filesystem is not validated to be a real ZFS filesystem,
+# since we choose not to care if the user locks a non-existent filesystem.
+#
+# There is no validation that the unlocker is the one holding the lock.
+#
+# Author: Simon Guest, 22/9/2014
+# Licensed under GNU General Public License GPLv3
+
+import optparse
+import os
+import os.path
+from stat import ST_MTIME
+import sys
+import time
+from zfstools.util import stderr, verbose_stderr, set_verbose
+
+LOCKDIR = "/var/lib/zfs-tools/zlock"
+
+def ensure_lockdir_exists():
+    if not os.path.isdir(LOCKDIR):
+        os.makedirs(LOCKDIR, 0755)
+
+def filesystem2lockfile(filesystem):
+    return filesystem.replace('/', ';')
```

Comment:
These two functions ought to use reversible encoding like urllib.  Alternatively the systemd-escape program provides reversible encoding.